redfish-core/lib/cable.hpp

```diff
@@ -60,17 +99,236 @@ inline void
+            std::move(linkArray);
+    });
+}
+
+inline void doGetCableDownstreamResources(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& cableObjectPath,
+    const std::vector<std::string>& updatedAssemblyList)
+{
+    std::string downstreamResource = cableObjectPath + "/downstream_resource";
+    linkAssociatedCable(asyncResp, downstreamResource,
+                        [asyncResp, cableObjectPath, updatedAssemblyList](
+                            const dbus::utility::MapperEndPoints& endpoints) {
+        nlohmann::json::array_t linkArray;
+        for (const auto& fullPath : endpoints)
+        {
+            auto it = std::ranges::find(updatedAssemblyList.begin(),
+                                        updatedAssemblyList.end(), fullPath);
+
+            // If element was found
+            if (it == updatedAssemblyList.end())
+            {
+                BMCWEB_LOG_ERROR(
+                    "in Downstream Resources {} isn't found in chassis assembly list",
+                    fullPath);
+                continue;
+            }
+            uint index = static_cast<uint>(it - updatedAssemblyList.begin());
+
+            nlohmann::json::object_t item;
+            item["@odata.id"] = boost::urls::format(
+                "/redfish/v1/Chassis/chassis/Assembly#/Assemblies/{}",
+                std::to_string(index));
+            linkArray.emplace_back(item);
+        }
+        asyncResp->res.jsonValue["Links"]["DownstreamResources@odata.count"] =
+            linkArray.size();
+        asyncResp->res.jsonValue["Links"]["DownstreamResources"] =
+            std::move(linkArray);
+    });
+}
+
+inline void getCableDownstreamResources(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& cableObjectPath)
+{
+    // retrieve Downstream Resources
+    getChassisAssembly(
+        asyncResp, "chassis",
+        [asyncResp,
+         cableObjectPath](const std::optional<std::string>& validChassisPath,
+                          const std::vector<std::string>& updatedAssemblyList) {
+        if (!validChassisPath || updatedAssemblyList.empty())
+        {
+            BMCWEB_LOG_DEBUG("Chassis not found");
+            return;
+        }
+        doGetCableDownstreamResources(asyncResp, cableObjectPath,
+                                      updatedAssemblyList);
+    });
+}
+
+inline void
+    getCableAssociatedPorts(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                            const nlohmann::json::json_pointer& jsonKeyName,
+                            const std::string& cableObjectPath,
+                            const std::string& associationName)
+{
+    sdbusplus::message::object_path endpointPath{cableObjectPath};
+    endpointPath /= associationName;
+    linkAssociatedCable(asyncResp, endpointPath,
+                        [asyncResp, jsonKeyName](
+                            const dbus::utility::MapperEndPoints& endpoints) {
```

Comment:
Should this > 10 line lambda be in a named function?

Reply:
I refactored those lambda usage cases.